drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -1827,7 +1827,7 @@ function DragAndDropBlock(runtime, element, configuration) {
     };
 
     var canSubmitAttempt = function() {
-        return Object.keys(state.items).length > 0 && attemptsRemain() && !submittingLocation();
+        return Object.keys(state.items).length > 0 && problemNotDue() && attemptsRemain() && !submittingLocation();
```

Comment:
Shouldn't this be enforced on the server as well? Otherwise students can just change their local clock and then submit, or run `const pastTime = (new Date).getTime() - 1e12; Date.prototype.getTime = () => pastTime;` in the browser console and then submit.

Reply:
Nice suggestion. I am thinking of moving all this logic to the server and just provide with a flag on the JS side.

Reply:
@bradenmacdonald The problem with user changing JS param is solely for the due dates check, but wherever we are making checks, user can do the manipulation to enable/disable certain buttons. For example, I was able to manipulate the max attempts and enable the `Show Answer` button. So, even if I do all the checks on the server side, any user can change that flag using browser consoles. For other problems, we are doing such checks in the python and rendering HTML based on the context. Based on what I have tested at the moment, it is very easy for a user to manipulate and change the behavior of the DnD component. 

![dnd_attempts_bug](https://user-images.githubusercontent.com/40599381/61104144-96926a80-a48e-11e9-9ac0-36ca0c3402c8.png)